action-allowedlist/src/allowed-action-analyzer.psm1

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+function Invoke-ValidateActions {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]
+        $approvedPath = "/src/approved.json",
+
+        [Parameter()]
+        [System.Object]
+        $ActionsConfiguration
+    )
+
+    Write-InfoLog "Checking if used actions are approved"
```

Comment:
While testing, I discovered something that is not very clearly documented in PowerShell:

1. The `Write-Host` command in a module _will_ write to the console,
2. But PowerShell formally recommends against using `Write-Host`, and
3. The replacement functions _do not write to the console_ when called from a module!

As a replacement, I created a custom logger.